File: app/src/main/java/mx/dev/franco/automusictagfixer/utilities/shared_preferences/AbstractSharedPreferences.java
package mx.dev.franco.automusictagfixer.utilities.shared_preferences;

public abstract class AbstractSharedPreferences {
    public abstract void putInt(String key, int value);
    public abstract void putString(String key, String value);
    public abstract void putLong(String key, long value);
    public abstract void putFloat(String key, float value);
    public abstract void putBoolean(String key, boolean value);
    public abstract int getInt(String key);
    public abstract String getString(String key);
    public abstract long getLong(String key);
    public abstract float getFloat(String key);
    public abstract boolean getBoolean(String key);
    public abstract void remove(String key);
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/utilities/shared_preferences/AbstractSharedPreferences.kt
package mx.dev.franco.automusictagfixer.utilities.shared_preferences


abstract class AbstractSharedPreferences {
    abstract fun putInt(key: String?, value: Int)
    abstract fun putString(key: String?, value: String?)
    abstract fun putLong(key: String?, value: Long)
    abstract fun putFloat(key: String?, value: Float)
    abstract fun putBoolean(key: String?, value: Boolean)
    abstract fun getInt(key: String?): Int
    abstract fun getString(key: String?): String?
    abstract fun getLong(key: String?): Long
    abstract fun getFloat(key: String?): Float
    abstract fun getBoolean(key: String?): Boolean
    abstract fun remove(key: String?)
}